File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    
        
    int answer = 1;
    map<string, int> mp;
    
    
    for(int i = 0 ;i<clothes.size();i++){
        if(mp.find(clothes[i][1])==mp.end())mp.insert({clothes[i][1],1});
        else  mp[clothes[i][1]]++;
    }
    
    for (const auto& [key, val] : mp) {
        answer *= (val + 1);
    }
    
    answer -= 1;
    
    return answer;
}

File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    clo_list = defaultdict(list)
    for i in range(len(clothes)):
        clo_list[clothes[i][1]].append(clothes[i][0])
        
    for key in clo_list:
        answer *= len(clo_list[key]) + 1
        
    answer -= 1
    return answer